src/pip/_internal/resolution/resolvelib/provider.py

```diff
@@ -115,11 +115,18 @@ def get_preference(
         self,
         resolution,  # type: Optional[Candidate]
         candidates,  # type: Sequence[Candidate]
-        information  # type: Sequence[Tuple[Requirement, Candidate]]
+        information  # type: Sequence[Tuple[Requirement, Optional[Candidate]]]
```

Comment:
Part of me wonders if we could call this something else, since "information" is too generic. OTOH, `dependency_information` seems even worse tho. ('.')

Reply:
`information` is what resolvelib calls it. Agreed that a better name might be nice, but let's leave that as a resolvelib question :-)